network/netcore/src/transport/tcp.rs

```diff
@@ -93,14 +94,17 @@ impl Transport for TcpTransport {
     }
 
     fn dial(&self, addr: NetworkAddress) -> Result<Self::Outbound, Self::Error> {
-        let (addr_string, _addr_suffix) =
-            parse_addr_and_port(addr.as_slice()).ok_or_else(|| invalid_addr_error(&addr))?;
+        let protos = addr.as_slice();
```

Comment:
qq: What does protos stand for at this point?

Reply:
`NetworkAddress` is "just" a list of `Protocol`s (really a `Vec<Protocol>`), so `NetworkAddress::as_slice()` gives us a slice into the list of protocols, hence "protos" for short.